src/database/schema/ChatSchema.ts

```diff
@@ -354,29 +366,67 @@ class ChatSchema implements BaseDbSchema {
     }
   };
 
-  static async deleteChat(localDb: SQLiteDatabase, messageId: string): Promise<ChatSchema> {
-    const existingChat = await ChatSchema.getByid(localDb, messageId);
-    return new ChatSchema({
-      ...existingChat,
-      type: 'deleted',
-      message: 'This message has been deleted'
-    });
-  }
-
   static updateDeletedChatType = async (db: SQLiteDatabase, messageId: string) => {
     try {
+      const selectQuery = `SELECT raw_json FROM ${ChatSchema.getTableName()} WHERE id = ?;`;
+      const resultSet = await db.executeSql(selectQuery, [messageId]);
+      const result = resultSet[0].rows.item(0);
+      const rawJson = JSON.parse(result.raw_json);
+      rawJson.message_type = MESSAGE_TYPE_DELETED;
+      rawJson.text = DELETED_MESSAGE_TEXT;
+      const updatedRawJson = JSON.stringify(rawJson);
+
       const updateQuery = `UPDATE ${ChatSchema.getTableName()}
-        SET type = ?, message = ?
+        SET type = ?, message = ?, raw_json = ?
         WHERE id = ?;`;
 
-      const updateReplacement = ['deleted', 'This message has been deleted', messageId];
+      const updateReplacement = ['deleted', DELETED_MESSAGE_TEXT, updatedRawJson, messageId];
 
       await db.executeSql(updateQuery, updateReplacement);
     } catch (e) {
       console.log('error updating deleted chat:', e);
     }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `updateDeletedChatType` method has been significantly modified. It now fetches the existing chat message, updates its `message_type` and `text` fields in the `raw_json` column, and then updates the chat message in the database. This change seems fine as long as the `raw_json` column is expected to contain a JSON string that can be parsed into an object with `message_type` and `text` fields.
```diff
-  static async deleteChat(localDb: SQLiteDatabase, messageId: string): Promise<ChatSchema> {
-    const existingChat = await ChatSchema.getByid(localDb, messageId);
-    return new ChatSchema({
-      ...existingChat,
-      type: 'deleted',
-      message: 'This message has been deleted'
-    });
-  }
-
-   static updateDeletedChatType = async (db: SQLiteDatabase, messageId: string) => {
-     try {
-       const updateQuery = `UPDATE ${ChatSchema.getTableName()}
-        SET type = ?, message = ?
-         WHERE id = ?;`;
- 
-      const updateReplacement = ['deleted', 'This message has been deleted', messageId];
- 
-       await db.executeSql(updateQuery, updateReplacement);
-     } catch (e) {
-       console.log('error updating deleted chat:', e);
-     }
-   };
+   static updateDeletedChatType = async (db: SQLiteDatabase, messageId: string) => {
+     try {
+       const selectQuery = `SELECT raw_json FROM ${ChatSchema.getTableName()} WHERE id = ?;`;
+       const resultSet = await db.executeSql(selectQuery, [messageId]);
+       const result = resultSet[0].rows.item(0);
+       const rawJson = JSON.parse(result.raw_json);
+       rawJson.message_type = MESSAGE_TYPE_DELETED;
+       rawJson.text = DELETED_MESSAGE_TEXT;
+       const updatedRawJson = JSON.stringify(rawJson);
+ 
+       const updateQuery = `UPDATE ${ChatSchema.getTableName()}
+        SET type = ?, message = ?, raw_json = ?
+         WHERE id = ?;`;
+ 
+      const updateReplacement = ['deleted', DELETED_MESSAGE_TEXT, updatedRawJson, messageId];
+ 
+       await db.executeSql(updateQuery, updateReplacement);
+     } catch (e) {
+       console.log('error updating deleted chat:', e);
+     }
+   };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->